messaging/Kafka-producer/src/test/java/io/quarkus/ts/messaging/kafka/BlockingProducerIT.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.ts.messaging.kafka;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.Map;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.helpers.test.UniAssertSubscriber;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+@Tag("QUARKUS-1090")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@QuarkusScenario
+public class BlockingProducerIT {
+    private static final int TIMEOUT_SEC = 5;
+    private static final int EVENTS = 100;
+    static final long KAFKA_MAX_BLOCK_MS = 100;
+    static final long DEVIATION_ERROR_MS = 80;
+    static final long KAFKA_MAX_BLOCK_TIME_MS = KAFKA_MAX_BLOCK_MS + DEVIATION_ERROR_MS;
+
+    static CustomStrimziKafkaContainer kafkaContainer;
```

Comment:
What limitation did you find to not use the standard `KafkaContainer`? If it's relevant, we should include it in the KafkaContainer from the test framework.
Pls, create an issue in the test framework and clarify what you are missing (and fix if you have time :D )

Reply:
quarkus-qe/quarkus-test-framework#248